HW10_API/API.py

```diff
@@ -0,0 +1,110 @@
+import requests
+import re
+from bs4 import BeautifulSoup
+
+
+class GenscanOutput:
+    def __init__(self, status, cds_list, intron_list, exon_list):
+        self.status = status
+        self.cds_list = cds_list
+        self.intron_list = intron_list
+        self.exon_list = exon_list
+
+
+def run_genscan(sequence=None, sequence_file=None, organism="Vertebrate",
+                exon_cutoff=1.00, sequence_name="", print_options='Predicted peptides only'):
+    '''
+    API for the GENSCAN Web Server at MIT
+    Identification of complete gene structures in genomic DNA
+
+        :param sequence: str with DNA sequence (default None)
+        :param sequence_file: path to file with DNA sequence (default None)
+        :param organism: one of the possible 'Vertebrate', 'Arabidopsis', 'Maize' (default 'Vertebrate')
+        :param exon_cutoff: one of the possible 1.00, 0.5, 0.25, 0.1, 0.05, 0.02, 0.01 (default 1.00)
+        :param sequence_name: str with name (default "")
+        :param print_options: 'Predicted peptides only' (default) or 'Predicted CDS and peptides'
+
+        :return: GenscanOutput object
+    '''
+
+    form_url = 'http://hollywood.mit.edu/cgi-bin/genscanw_py.cgi'
+
+    if sequence_file is not None:
+        sequence_file = open(sequence_file, "rb")
+        files = {'-u': sequence_file}
+
+        payload = {
+            '-o': organism,
+            '-e': exon_cutoff,
+            '-n': sequence_name,
+            '-p': print_options,
+        }
+
+        resp = requests.post(form_url, data=payload, files=files)
+        sequence_file.close()
```

Comment:
На самом деле никто не запрещает использовать with и в таких случаях, просто на лекции был не самый удачный пример

```suggestion
        with open(sequence_file, "rb") as sequence_file:
            files = {'-u': sequence_file}

            payload = {
            '-o': organism,
            '-e': exon_cutoff,
            '-n': sequence_name,
            '-p': print_options,
        }

            resp = requests.post(form_url, data=payload, files=files)
```